src/scenarios/shared/sod.py

```diff
@@ -0,0 +1,73 @@
+'''
+Wrapper around startup tool.
+'''
+import sys
+import os
+import platform
+from shutil import copytree
+from performance.logger import setup_loggers
+from performance.common import get_artifacts_directory, get_packages_directory, RunCommand
+from performance.constants import UPLOAD_CONTAINER, UPLOAD_STORAGE_URI, UPLOAD_TOKEN_VAR, UPLOAD_QUEUE
+from dotnet import CSharpProject, CSharpProjFile
+from shared.util import helixpayload, helixworkitempayload, helixuploaddir, builtexe, publishedexe, runninginlab, uploadtokenpresent, getruntimeidentifier
+from shared.const import *
+class SODWrapper(object):
+    '''
+    Wraps sod.exe, building it if necessary.
+    '''
+    def __init__(self):
+        if helixpayload() and os.path.exists(os.path.join(helixpayload(), 'SOD')):
+            self._setsodpath(os.path.join(helixpayload(), 'SOD'))
+        elif helixworkitempayload() and os.path.exists(os.path.join(helixworkitempayload(), 'SOD')):
+            self._setsodpath(os.path.join(helixworkitempayload(), 'SOD'))
+        else:
+            relpath = os.path.join(get_artifacts_directory(), 'SOD')
+            startupproj = os.path.join('..',
+                                       '..',
+                                       'tools',
+                                       'ScenarioMeasurement',
+                                       'SizeOnDisk',
+                                       'SizeOnDisk.csproj')
+            sod = CSharpProject(CSharpProjFile(startupproj,
+                                                   sys.path[0]),
+                                                   os.path.join(os.path.dirname(startupproj),
+                                    os.path.join(get_artifacts_directory(), 'SOD')))
+            if not os.path.exists(relpath):
+                sod.restore(get_packages_directory(),
+                                True,
+                                getruntimeidentifier())
+                sod.publish('Release',
+                                relpath,
+                                True,
+                                get_packages_directory(),
+                                None,
+                                getruntimeidentifier(),
+                                '--no-restore'
+                                )
+            self._setsodpath(sod.bin_path)
+
+    
+    def _setsodpath(self, path: str):
+        self.sodexe = os.path.join(path, 'SizeOnDisk.exe')
+
+    def runtests(self, scenarioname, dirs):
+        '''
+        Runs tests through sod tool
+        '''
+        if not os.path.exists(TRACEDIR):
+            os.mkdir(TRACEDIR)
+        reportjson = os.path.join(TRACEDIR, 'perf-lab-report.json')
```

Comment:
Should the name of the json have more information about the test?

Reply:
I didn't do that because they get persisted to individual locations with the job and workitem GUID which should be enough to avoid dupes, and the test info is inside the json. (This decision predates this PR.)